app/src/test/kotlin/social/plasma/relay/EventSerdeTest.kt

```diff
@@ -0,0 +1,54 @@
+package social.plasma.relay
+
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
+import io.kotest.core.spec.style.StringSpec
+import io.kotest.matchers.shouldBe
+import io.kotest.property.Arb
+import io.kotest.property.arbitrary.*
+import io.kotest.property.checkAll
+import okio.ByteString
+import okio.ByteString.Companion.toByteString
+import social.plasma.relay.json.HexByteStringAdapter
+import social.plasma.relay.json.InstantAdapter
+import java.time.Instant
+import java.time.temporal.ChronoUnit
+
+class EventSerdeTest : StringSpec({
+
+    val subject = Moshi.Builder()
+        .add(InstantAdapter())
+        .add(HexByteStringAdapter())
+        .addLast(KotlinJsonAdapterFactory())
+        .build().adapter(Event::class.java)
+
+    "can serde events" {
+        checkAll(arbEvent) { event ->
```

Comment:
property based testing in kotest. This will generate 100 random events (using the Arb builders below) and test them.

I read that kotest will only work for JVM unit tests. Cannot test Android UI stuff this way.